helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -2061,4 +2070,45 @@ public ZKHelixAdmin build() {
               _zkAddress), false);
     }
   }
+
+  private Map<String, InstanceConfig> findTimeoutOfflineInstances(String clusterName,
+      Map<String, String> customizedPurgeMap) {
```

Comment:
From the API logic, it seems what is needed is only `timeout`? Then I think we don't have to pass along the map, but only the `timeout` value from the rest api layer: `timeout = customizedPurgeMap.get(OFFLINE_NODE_PURGE_TIMEOUT);`
Then the APIs only need the timeout param, not a map.